code/game/machinery/computer/slotmachine.dm

```diff
@@ -0,0 +1,296 @@
+	add_fingerprint(user)
+	if(stat & (BROKEN|NOPOWER))
+		return
+	ui_interact(user)
+
+/obj/machinery/computer/slot_machine/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/computer/slot_machine/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/weapon/coin))
+		var/obj/item/weapon/coin/C = I
+		to_chat(user, "<span class='notice'>You insert a \the [C] into \the [src]'s slot!</span>")
+		var/value = get_value(C)
+		balance += value
+		qdel(C)
+	else
+		return ..()
+
+/obj/machinery/computer/slot_machine/emag_act()
+	if(!emagged)
+		emmaged = TRUE
+		spark(src, 3)
+		playsound(src, "sparks", 50, 1)
+		return TRUE
+
+/obj/machinery/computer/slot_machine/ui_interact(mob/living/user)
+	. = ..()
+	var/reeltext = {"<center><font face=\"courier new\">
+	/*****^*****^*****^*****^*****\\<BR>
+	| \[[reels[1][1]]\] | \[[reels[2][1]]\] | \[[reels[3][1]]\] | \[[reels[4][1]]\] | \[[reels[5][1]]\] |<BR>
+	| \[[reels[1][2]]\] | \[[reels[2][2]]\] | \[[reels[3][2]]\] | \[[reels[4][2]]\] | \[[reels[5][2]]\] |<BR>
+	| \[[reels[1][3]]\] | \[[reels[2][3]]\] | \[[reels[3][3]]\] | \[[reels[4][3]]\] | \[[reels[5][3]]\] |<BR>
+	\\*****v*****v*****v*****v*****/<BR>
+	</center></font>"}
+
+	var/dat
+	if(working)
+		dat = reeltext
+
+	else
+		dat = {"Five credits to play!<BR>
+		<B>Prize Money Available:</B> [money] (jackpot payout is ALWAYS 100%!)<BR>
+		<B>Credit Remaining:</B> [balance]<BR>
+		[plays] players have tried their luck today, and [jackpots] have won a jackpot!<BR>
+		<HR><BR>
+		<A href='?src=\ref[src];spin=1'>Play!</A><BR>
+		<BR>
+		[reeltext]
+		<BR>
+		<font size='1'><A href='?src=\ref[src];refund=1'>Refund balance</font></A><BR>"}
+
+	var/datum/browser/popup = new(user, "slotmachine", "Slot Machine")
+	popup.set_content(dat)
+	popup.set_title_image(user.browse_rsc_icon(icon, icon_state))
+	popup.open()
+
+/obj/machinery/computer/slot_machine/Topic(href, href_list)
+	. = ..() //Sanity checks.
+	if(.)
+		return .
+
+	if(href_list["spin"])
+		spin(usr)
+
+	else if(href_list["refund"])
+		give_coins(balance)
+		balance = 0
+
+/obj/machinery/computer/slot_machine/emp_act(severity)
+	. = ..()
+	if(stat & (NOPOWER|BROKEN))
+		return
+	var/severity_ascending = 4 - severity
+	money = max(rand(money - (200 * severity_ascending), money + (200 * severity_ascending)), 0)
+	balance = max(rand(balance - (50 * severity_ascending), balance + (50 * severity_ascending)), 0)
+	money -= max(0, give_coins(min(rand(-50, 100 * severity_ascending)), money))
+	spin()
+
+/obj/machinery/computer/slot_machine/proc/spin(mob/user)
+	if(!can_spin(user))
+		return
+
+	var/the_name
+	if(user)
+		the_name = user.real_name
+		visible_message("<span class='notice'>[user] pulls the lever and the slot machine starts spinning!</span>")
+	else
+		the_name = "Exaybachay"
+
+	balance -= SPIN_PRICE
+	money += SPIN_PRICE
+	plays += 1
+	working = 1
+
+	toggle_reel_spin(1)
+	update_icon()
+	updateUsrDialog()
+
+	spawn(0)
```

Comment:
The headdevs will probably tell you to convert all the spawns to timers so I'm just doing this suggestion as a warning.